server/sonar-server/src/main/java/org/sonar/server/organization/ws/UpdateProjectVisibilityAction.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.organization.ws;
+
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+
+import static org.sonar.server.organization.ws.OrganizationsWsSupport.PARAM_ORGANIZATION;
+
+public class UpdateProjectVisibilityAction implements OrganizationsWsAction {
+  static final String ACTION = "update_project_visibility";
+  static final String PARAM_PROJECT_VISIBILITY = "projectVisibility";
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION)
+      .setPost(true)
+      .setDescription("Update the default visibility for new projects of the specified organization.")
+      .setInternal(true)
+      .setSince("6.4")
+      .setHandler(this);
+
+    action.createParam(PARAM_ORGANIZATION)
+      .setRequired(true)
+      .setDescription("Organization key")
+      .setExampleValue("foo-company");
+
+    action.createParam(PARAM_PROJECT_VISIBILITY)
+      .setRequired(true)
+      .setDescription("Default visibility for projects")
+      .setPossibleValues("private", "public");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
```

Comment:
As discussed I'll squash the last two commits, to avoid this invalid intermediate state.